pkg/factory/risingwave_object_factory_test.go

```diff
@@ -663,6 +663,24 @@ func listContains[T comparable](a, b []T) bool {
 	return true
 }
 
+func listContainsWithEquals[T any](a, b []T, equals func(x, y T) bool) bool {
```

Comment:
I don't get this function. This only returns true if all elements are equal. Is that what you want to do? 

```go
a := []int{1, 2, 3}
b := []int{1, 2, 3}
fmt.Println(listContainsWithEquals(a, b, comp)) // diff elements, false
a = []int{1}
b = []int{1, 2}
fmt.Println(listContainsWithEquals(a, b, comp)) // diff len, false
a = []int{1, 1}
b = []int{1, 1}
fmt.Println(listContainsWithEquals(a, b, comp)) // all same elements, false
```

Reply:
Just in case you wanted to write something like `check if all elements of a are in b`, the function could look something like this https://go.dev/play/p/UthfW4uUzPF

Reply:
Thanks, you are right. There was a bug. I'll fix it.